pkg/filter/network/streamproxy/streamproxy.go

```diff
@@ -74,12 +75,16 @@ func NewProxy(ctx context.Context, config *v2.StreamProxy, net string) Proxy {
 
 func (p *proxy) OnData(buffer buffer.IoBuffer) api.FilterStatus {
 	if log.DefaultLogger.GetLogLevel() >= log.DEBUG {
-		log.DefaultLogger.Debugf("[%s proxy] [ondata] read data , len = %v", p.network, buffer.Len())
+		log.DefaultLogger.Debugf("[%s proxy] [ondata] read data , len = %v, local addr:%s, upstream addr:%s",
+			p.network, buffer.Len(), p.readCallbacks.Connection().LocalAddr().String(), p.upstreamConnection.RemoteAddr().String())
 	}
 	bytesRecved := p.requestInfo.BytesReceived() + uint64(buffer.Len())
 	p.requestInfo.SetBytesReceived(bytesRecved)
 
-	p.upstreamConnection.Write(buffer.Clone())
+	err := p.upstreamConnection.Write(buffer.Clone())
+	if err == types.ErrConnectionHasClosed {
```

Comment:
no need to call onUpstreamEvent here, because Write error would call onUpstreamEvent directly.